### ISSUE TITLE
Expected substring `RE_EXN_ID`

### ISSUE THREAD

**Issue by idkjs** [glennsl/bs-jest#84]
I was updating this repo on this fork -> [https://github.com/idkjs/rebase/tree/reason](https://github.com/idkjs/rebase/tree/reason) and got the following errors when I ran the tests. It would seem  that the expected response was correct but the formatting was wrong. I thought it might be a snapshot thing but I cant find any in the repo.

I guess it has to do with what is being passed to `Jest.assertion`:

On `toThrow`, the expected substring has to fields an is escaped. The recieved value has three fields and is not escaped. So the second value looks correct but it not being decoded right?

```sh
    Expected substring: "{\"RE_EXN_ID\":\"Rebase__Types.InvalidArgument/4\",\"_1\":\"\"}"
    Received value:     {"Error": [Error], "RE_EXN_ID": "Rebase__Types.InvalidArgument/4", "_1": "Array.range: ~step=0 would cause infinite loop"}
```

How would I address fixing this issue?

```sh
FAIL  __tests__/Seq_test.bs.js
  ● range - step 0

    expect(received).toThrow(expected)

    Expected substring: "{\"RE_EXN_ID\":\"Rebase__Types.InvalidArgument/4\",\"_1\":\"\"}"
    Received value:     {"Error": [Error], "RE_EXN_ID": "Rebase__Types.InvalidArgument/4", "_1": "Seq.range: ~step=0 would cause infinite loop"}

      at affirm (node_modules/@glennsl/bs-jest/src/jest.bs.js:153:38)
      at Object.<anonymous> (node_modules/@glennsl/bs-jest/src/jest.bs.js:238:11)

 FAIL  __tests__/Option_test.bs.js
  ● getOrRaise - None

    expect(received).toThrow(expected)

    Expected substring: "{\"RE_EXN_ID\":\"Rebase__Types.InvalidArgument/4\",\"_1\":\"getOrRaise called on None\"}"
    Received value:     {"Error": [Error], "RE_EXN_ID": "Rebase__Types.InvalidArgument/4", "_1": "getOrRaise called on None"}

      at affirm (node_modules/@glennsl/bs-jest/src/jest.bs.js:153:38)
      at Object.<anonymous> (node_modules/@glennsl/bs-jest/src/jest.bs.js:238:11)

 FAIL  __tests__/Result_test.bs.js
  ● wrap - Error

    expect(received).toEqual(expected) // deep equality

    - Expected  - 0
    + Received  + 1

      Object {
        "TAG": 1,
        "_0": Object {
    +     "Error": [Error],
          "RE_EXN_ID": "Failure",
          "_1": "err",
        },
      }

      at affirm (node_modules/@glennsl/bs-jest/src/jest.bs.js:82:38)
      at Object.<anonymous> (node_modules/@glennsl/bs-jest/src/jest.bs.js:238:11)

  ● wrap1 - Error

    expect(received).toEqual(expected) // deep equality

    - Expected  - 0
    + Received  + 1

      Object {
        "TAG": 1,
        "_0": Object {
    +     "Error": [Error],
          "RE_EXN_ID": "Failure",
          "_1": "err",
        },
      }

      at affirm (node_modules/@glennsl/bs-jest/src/jest.bs.js:82:38)
      at Object.<anonymous> (node_modules/@glennsl/bs-jest/src/jest.bs.js:238:11)

  ● wrap2 - Error

    expect(received).toEqual(expected) // deep equality

    - Expected  - 0
    + Received  + 1

      Object {
        "TAG": 1,
        "_0": Object {
    +     "Error": [Error],
          "RE_EXN_ID": "Failure",
          "_1": "err",
        },
      }

      at affirm (node_modules/@glennsl/bs-jest/src/jest.bs.js:82:38)
      at Object.<anonymous> (node_modules/@glennsl/bs-jest/src/jest.bs.js:238:11)

  ● getOrRaise - Error

    expect(received).toThrow(expected)

    Expected substring: "{\"RE_EXN_ID\":\"Rebase__Types.InvalidArgument/4\",\"_1\":\"getOrRaise called on Error\"}"
    Received value:     {"Error": [Error], "RE_EXN_ID": "Rebase__Types.InvalidArgument/4", "_1": "getOrRaise called on Error"}

      at affirm (node_modules/@glennsl/bs-jest/src/jest.bs.js:153:38)
      at Object.<anonymous> (node_modules/@glennsl/bs-jest/src/jest.bs.js:238:11)

 PASS  __tests__/Fn_test.bs.js
 FAIL  __tests__/List_test.bs.js
  ● range - step 0

    expect(received).toThrow(expected)

    Expected substring: "{\"RE_EXN_ID\":\"Rebase__Types.InvalidArgument/4\",\"_1\":\"\"}"
    Received value:     {"Error": [Error], "RE_EXN_ID": "Rebase__Types.InvalidArgument/4", "_1": "List.range: ~step=0 would cause infinite loop"}

      at affirm (node_modules/@glennsl/bs-jest/src/jest.bs.js:153:38)
      at Object.<anonymous> (node_modules/@glennsl/bs-jest/src/jest.bs.js:238:11)

 PASS  __tests__/ReasonSyntax_test.bs.js
 FAIL  __tests__/Array_test.bs.js
  ● range - step 0

    expect(received).toThrow(expected)

    Expected substring: "{\"RE_EXN_ID\":\"Rebase__Types.InvalidArgument/4\",\"_1\":\"\"}"
    Received value:     {"Error": [Error], "RE_EXN_ID": "Rebase__Types.InvalidArgument/4", "_1": "Array.range: ~step=0 would cause infinite loop"}

      at affirm (node_modules/@glennsl/bs-jest/src/jest.bs.js:153:38)
      at Object.<anonymous> (node_modules/@glennsl/bs-jest/src/jest.bs.js:238:11)

  ● getOrRaise - out of bounds

    expect(received).toThrow(expected)

    Expected substring: "{\"RE_EXN_ID\":\"Rebase__Types.IndexOutOfBounds/5\"}"
    Received value:     {"Error": [Error], "RE_EXN_ID": "Rebase__Types.IndexOutOfBounds/5"}

      at affirm (node_modules/@glennsl/bs-jest/src/jest.bs.js:153:38)
      at Object.<anonymous> (node_modules/@glennsl/bs-jest/src/jest.bs.js:238:11)

  ● setOrRaise - out of bounds

    expect(received).toThrow(expected)

    Expected substring: "{\"RE_EXN_ID\":\"Rebase__Types.IndexOutOfBounds/5\"}"
    Received value:     {"Error": [Error], "RE_EXN_ID": "Rebase__Types.IndexOutOfBounds/5"}

      at affirm (node_modules/@glennsl/bs-jest/src/jest.bs.js:153:38)
      at Object.<anonymous> (node_modules/@glennsl/bs-jest/src/jest.bs.js:238:11)

-----------------------|---------|----------|---------|---------|-------------------
File                   | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
-----------------------|---------|----------|---------|---------|-------------------
All files              |     100 |      100 |     100 |     100 |
 __tests__/helpers     |     100 |      100 |     100 |     100 |
  TestHelpers.bs.js    |     100 |      100 |     100 |     100 |
 src                   |     100 |      100 |     100 |     100 |
  Rebase.bs.js         |     100 |      100 |     100 |     100 |
  Rebase__Array.bs.js  |     100 |      100 |     100 |     100 |
  Rebase__Fn.bs.js     |     100 |      100 |     100 |     100 |
  Rebase__List.bs.js   |     100 |      100 |     100 |     100 |
  Rebase__Option.bs.js |     100 |      100 |     100 |     100 |
  Rebase__Result.bs.js |     100 |      100 |     100 |     100 |
  Rebase__Seq.bs.js    |     100 |      100 |     100 |     100 |
  Rebase__String.bs.js |     100 |      100 |     100 |     100 |
  Rebase__Types.bs.js  |     100 |      100 |     100 |     100 |
-----------------------|---------|----------|---------|---------|-------------------
Test Suites: 5 failed, 3 passed, 8 total
Tests:       10 failed, 323 passed, 333 total
Snapshots:   0 total
Time:        4.746 s
Ran all test suites.
```

Thank you,sir.

**Comment by glennsl** [comment]
Hey. toThrowException`, `toThrowMessage` and `toThrowMessageRe` were removed in 0.6 because they relied on assumptions about BuckleScript internals that no longer hold. To upgrade to 0.6 you'd have to use `toThrow` instead, which isn't as accurate, but should be sufficient in most cases. In the cases where it's not I think you'll just have to catch the exceptions yourself, pattern match on them and return the basic `pass`/`fail` assertions.

**Comment by glennsl** [comment]
Hmm, apparently I just removed the tests, not the actual expects. The rest of my comment should be accurate though!

**Comment by glennsl** [comment]
Just released 0.7, which actually removes the expects. If you upgrade to that it'll fail in the right way now!

**Comment by idkjs** [comment]
https://github.com/glennsl/bs-jest/issues/84#issuecomment-798904234 resolves the issue. Thank you, sir.
